MapboxNavigation/RouteMapViewController.swift

```diff
@@ -315,9 +317,15 @@ extension RouteMapViewController: NavigationMapViewDelegate {
     func navigationMapView(_ mapView: MGLMapView, imageFor annotation: MGLAnnotation) -> MGLAnnotationImage? {
         return delegate?.navigationMapView(mapView, imageFor:annotation)
     }
+    
+    func mapViewDidFinishLoadingMap(_ mapView: MGLMapView) {
+        hasFinishedLoadingStyle = true
+    }
 
     @objc(navigationMapView:shouldUpdateTo:)
     func navigationMapView(_ mapView: NavigationMapView, shouldUpdateTo location: CLLocation) -> CLLocation? {
+        
+        guard hasFinishedLoadingStyle else { return nil }
```

Comment:
If the style hasn’t finished loading, we can’t manipulate the style in order to query it for street names, but we should still run the rest of this method in order to snap the user’s location to the route.

In fact, that’s what the code already does: it [checks for a non-`nil` `style`](https://github.com/mapbox/mapbox-navigation-ios/blob/eab11c929885b270e8c655495ed4e0b25123a78b/MapboxNavigation/RouteMapViewController.swift#L335) before manipulating the style. `style` would be `nil` until the style has finished loading.